2012-02-08  Jeff Law  <law@redhat.com>

        * sysdeps/unix/sysv/linux/x86_64/pthread_cond_wait.S: fix lost
	  wakeups.

---
 nptl/sysdeps/unix/sysv/linux/x86_64/pthread_cond_wait.S |    3 +++
 1 file changed, 3 insertions(+)

--- a/nptl/sysdeps/unix/sysv/linux/x86_64/pthread_cond_wait.S
+++ b/nptl/sysdeps/unix/sysv/linux/x86_64/pthread_cond_wait.S
@@ -365,6 +365,9 @@
 	incl	cond_futex(%rdi)
 	movl	cond_futex(%rdi), %edx
 
+	/* Increment total_seq to ensure we do not lose wakeups.  */
+	incq	total_seq(%rdi)
+
 	/* Release internal lock.  */
 	LOCK
 #if cond_lock == 0
